user/serializers.py

```diff
@@ -22,3 +23,33 @@ def update(self, instance, validated_data):
             user.save()
 
         return user
+
+
+class AuthTokenSerializer(serializers.Serializer):
+    email = serializers.CharField(label=_("Email"))
+    password = serializers.CharField(
+        label=_("Password"), style={"input_type": "password"}
+    )
+
+    def validate(self, attrs):
+        email = attrs.get("email")
+        password = attrs.get("password")
+
+        if email and password:
+            user = authenticate(email=email, password=password)
+
+            if user:
+                if not user.is_active:
+                    msg = _("User account is disabled.")
+                    raise serializers.ValidationError(
+                        msg, code="authorization"
+                    )
+            else:
+                msg = _("Unable to log in with provided credentials.")
+                raise serializers.ValidationError(msg, code="authorization")
+        else:
+            msg = _("Must include 'username' and 'password'.")
+            raise serializers.ValidationError(msg, code="authorization")
```

Comment:
this can be simplified a bit:)
start from `if not user`

Reply:
got it!